nemo/collections/common/data/lhotse/nemo_adapters.py

```diff
@@ -274,10 +322,13 @@ def shard_ids(self) -> List[int]:
     def __iter__(self) -> Generator[Cut, None, None]:
         shard_ids = self.shard_ids
 
+        seed = resolve_seed(self.shard_seed)
         if self.shuffle_shards:
-            seed = resolve_seed(self.shard_seed)
             random.Random(seed).shuffle(shard_ids)
 
+        # Propagate the random seed
```

Comment:
just for reproducibility or is there any other reason?

Reply:
Reproducibility/consistent RNG behavior across all dataloading modules.